src/modules/nft/src/index.js

```diff
@@ -0,0 +1,105 @@
+import configs from './config';
+import API from './api';
+import NftCollection from './nftCollection';
+
+export default class Nft {
+  constructor(environment = {}) {
+    this.active = true;
+    this.network = environment.network || { type: { name: 'ETH' } };
+    this.activeAddress = environment.address;
+    this.tokenSetUpdateHook = environment.tokenSetUpdateHook;
+    this.setAvailableContracts = environment.setAvailableContracts;
+    this.errorHandler = environment.errorHandler || console.error;
+    this.web3 = environment.web3;
+    this.nftUrl = `${configs.url}getImage`;
+    this.openSeaLambdaUrl = configs.url;
+    this.nftConfig = {};
+    this.selectedContract = '';
+    this.detailsFor = {};
+    this.ownedTokenBasicDetails = [];
+
+    this.api = new API({
+      url: this.openSeaLambdaUrl,
+      address: this.activeAddress,
+      errorHandler: this.errorHandler
+    });
+  }
+
+  static async init(environment) {
+    const create = new Nft(environment);
+    await create.setup();
+    return create;
+  }
+
+  //======================================================================
+  setSelectedContract(selectedContract) {
+    this.selectedContract = selectedContract;
+  }
+
+  getOwnedTokenBasicDetails() {
+    return this.ownedTokenBasicDetails;
+  }
+
+  hasImage(nft) {
+    if (nft.customNft || nft.image === '') {
+      return true;
+    }
+    if (nft.loaded) {
+      return true;
+    }
+  }
+
+  async setup() {
+    const nftData = {};
+    let selectedContract;
+    if (this.network.type.name === 'ETH') {
+      const configData = await this.api.getTokens();
+      if (!configData.error) {
+        try {
+          configData.tokenContracts.forEach(data => {
+            this.ownedTokenBasicDetails.push({
+              name: data.name,
+              count: data.owned_asset_count,
+              contract: data.contractIdAddress
+            });
+            nftData[data.contractIdAddress] = new NftCollection({
+              details: data,
+              api: this.api,
+              address: this.activeAddress,
+              tokenSetUpdateHook: this.tokenSetUpdateHook,
+              web3: this.web3
+            });
+          });
+          this.nftConfig = { ...nftData };
+          selectedContract = Object.keys(this.nftConfig)[0];
+          this.selectedContract = Object.keys(this.nftConfig)[0];
+          this.setAvailableContracts(Object.keys(this.nftConfig));
+          return selectedContract;
+        } catch (e) {
+          this.errorHandler(e);
+        }
+      }
```

Comment:
dont hardcode `ETH` get it from networks
also there is no else?
you have to return Promise.resolve() or reject on else,
also remove awaits